### PR TITLE
Select iterator class.

### LINKED ISSUE #1: Select Iterator
Add SelectIterator class which takes a callback and processes rows in chunks with iterator.

### PR DESCRIPTION
**Pull request by mateuszherczka**
- closes #1 

SelectIterator calls callback(rowset_generator) once. Callback handles row generator.

### CODE REVIEW

File: src/dbquery/query.py [hunk 1]
```diff
@@ -178,6 +178,116 @@ def _produce_return(self, cursor):
...
+        self.callback = callback
+        self.cb_args = cb_args
+
+    def _rowset_generator(self, cursor):
+        """ Yields sets of rows with length == arraysize until no more rows
+        exist in query result. Applies row formatter.
+        """
+        while True:
+            rowset = cursor.fetchmany(self._arraysize)
+            if not rowset:
```
Comment (merry-bits) [inline src/dbquery/query.py]: Move into the while condition, using the fact that end of data is `[]`.

File: src/dbquery/query.py [hunk 2]
```diff
@@ -178,6 +178,116 @@ def _produce_return(self, cursor):
...
+                break
+            else:
+                if self._row_formatter is not None:
+                    rowset = (self._row_formatter(r, cursor) for r in rowset)
+                yield rowset
```
Comment (merry-bits) [inline src/dbquery/query.py]: Yield rows not a rowset.

File: src/dbquery/query.py [hunk 3]
```diff
@@ -178,6 +178,116 @@ def _produce_return(self, cursor):
 
         return row
 
+class SelectSetwise(Select):
```
Comment (merry-bits) [inline src/dbquery/query.py]: Not needed.

File: src/dbquery/query.py [hunk 4]
```diff
@@ -178,6 +178,116 @@ def _produce_return(self, cursor):
...
+        """
+        for rowset in self._rowset_generator(cursor):
+            if self._row_formatter is not None:
+                rowset = (self._row_formatter(r, cursor) for r in rowset)
+            self.callback(rowset, *self.cb_args)
+
+        return None
+
+
+class SelectGen(Select):
```
Comment (merry-bits) [inline src/dbquery/query.py]: Perhaps rename?

File: src/dbquery/query.py (hunk 1 above)
Comment (mateuszherczka) [reply src/dbquery/query.py]: done

File: src/dbquery/query.py (hunk 2 above)
Comment (mateuszherczka) [reply src/dbquery/query.py]: done

File: src/dbquery/query.py (hunk 3 above)
Comment (mateuszherczka) [reply src/dbquery/query.py]: nuked

File: src/dbquery/query.py (hunk 4 above)
Comment (mateuszherczka) [reply src/dbquery/query.py]: SelectIterator

**Comment by mateuszherczka** [comment]
Closed because we use pull request to original repo instead.

**Pull request closed without merging**